wait-for-plugins-and-start.sh

```diff
@@ -3,34 +3,57 @@
 # This script is a workaround for podman-compose absence of support for depends_on
 
 # Entrypoint for the main RHDH container.
-# Waits for the dynamic plugins config to be generated,
-# then starts the Backstage backend with optional local overrides.
+# Waits for the dynamic plugin config to be generated,
+# then starts the Backstage backend with appropriate config files.
+#
+# If user-supplied override files for catalog entities (users/components) exist,
+# this script replaces their paths in the base config accordingly.
 
 
 DYNAMIC_PLUGINS_CONFIG="dynamic-plugins-root/app-config.dynamic-plugins.yaml"
-USER_APP_CONFIG="configs/app-config/app-config.local.yaml"
 DEFAULT_APP_CONFIG="configs/app-config/app-config.yaml"
+PATCHED_APP_CONFIG="generated/app-config.patched.yaml"
+
+USER_APP_CONFIG="configs/app-config/app-config.local.yaml"
+LEGACY_USER_APP_CONFIG="configs/app-config.local.yaml"
+
+USERS_OVERRIDE="configs/catalog-entities/users.override.yaml"
+COMPONENTS_OVERRIDE="configs/catalog-entities/components.override.yaml"
+
+mkdir -p generated
+cp -f "$DEFAULT_APP_CONFIG" "$PATCHED_APP_CONFIG"
 
-# Wait for dynamic plugins config to be generated
+# Wait until the dynamic plugin config is ready
 while [ ! -f "$DYNAMIC_PLUGINS_CONFIG" ]; do
-    echo "Waiting for $DYNAMIC_PLUGINS_CONFIG to be created by install-dynamic-plugins container ..."
-    sleep 2
+  echo "Waiting for $DYNAMIC_PLUGINS_CONFIG to be created by install-dynamic-plugins container ..."
+  sleep 2
 done
 
-# Optionally include user app-config.local.yaml if it exists
+# Apply overrides by replacing target paths in the patched config
+if [ -f "$USERS_OVERRIDE" ]; then
+  echo "Applying users override"
+  sed -i "s|target: /opt/app-root/src/configs/catalog-entities/users.yaml|target: /opt/app-root/src/$USERS_OVERRIDE|" "$PATCHED_APP_CONFIG"
```

Comment:
```suggestion
  sed -i "s|/opt/app-root/src/configs/catalog-entities/users.yaml|/opt/app-root/src/$USERS_OVERRIDE|" "$PATCHED_APP_CONFIG"
```

I'd suggest replacing all occurrences, not just the `target: ` ones.